charts/trino/templates/configmap-coordinator.yaml

```diff
@@ -142,29 +122,26 @@ data:
   {{- end }}
 {{ end }}
 
-{{- range $fileName, $fileContent := .Values.coordinator.additionalConfigFiles }}
+{{- if .Values.accessControl }}
+  {{ include "trino.accessControlProperties" . }}
+{{- end }}
+
+{{- range $fileName, $fileContent := .Values.worker.additionalConfigFiles }}
   {{ $fileName }}: |
     {{- $fileContent | nindent 4 }}
 {{- end }}
 
 ---
 
-{{- if .Values.accessControl }}{{- if eq .Values.accessControl.type "configmap" }}
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  name: {{ template "trino.fullname" . }}-access-control-volume-coordinator
-  namespace: {{ .Release.Namespace }}
-  labels:
-    {{- include "trino.labels" . | nindent 4 }}
-    app.kubernetes.io/component: coordinator
-data:
-  {{- range $key, $val := .Values.accessControl.rules }}
-  {{ $key }}: {{ $val | quote }}
+{{- if .Values.accessControl }}
+  {{- if eq .Values.accessControl.type "configmap" }}
+  {{ include "trino.accessControlConfigMap" . }}
   {{- end }}
-{{- end }}{{- end }}
-{{- if .Values.resourceGroups }}
+{{- end }}
+
 ---
+{{- if .Values.resourceGroups }}
```

Comment:
Keep the doc separator (`---`) inside the condition block to avoid creating empty docs.